index.js

```diff
@@ -65,3 +67,125 @@ module.exports = (emitter, event, options) => {
 
 	return ret;
 };
+module.exports.iterator = (emitter, event, options) => {
+	if (typeof options === 'function') {
+		options = {filter: options};
+	}
+
+	options = Object.assign({
+		rejectionEvents: ['error'],
+		resolutionEvents: [],
+		multiArgs: false
+	}, options);
+
+	let addListener = emitter.on || emitter.addListener || emitter.addEventListener;
+	let removeListener = emitter.off || emitter.removeListener || emitter.removeEventListener;
+
+	if (!addListener || !removeListener) {
+		throw new TypeError('Emitter is not compatible');
+	}
+
+	addListener = addListener.bind(emitter);
+	removeListener = removeListener.bind(emitter);
```

Comment:
Line 81 to 89 could be shared with the main export implementation by abstracting it into a method that accepts the emitter and returns the `addListener` and `removeLister` methods.